File: IUTBeans/IUTBeans/Solution-2/Condiments/WhippedCream.cs
namespace IUTBeans.Solution_2.Condiments;

public class WhippedCream:ICondiment
{
    private ICondiment? _condiment;

    public WhippedCream(ICondiment? condiment)
    {
        _condiment = condiment;
    }
    public string GetDescription()
    {
        return "with Whipped cream " + (_condiment?.GetDescription() ?? ".");
    }

    public double GetPrice()
    {
        return 75 + (_condiment?.GetPrice() ?? 0);
    }
}

File: Lab_4/src/Condiment/WhippedCream.java
package Condiment;

public class WhippedCream implements ICondiment {

    private ICondiment nextCondiment;

    public WhippedCream(ICondiment nextCondiment){
        this.nextCondiment = nextCondiment;
    }

    public double getPrice(){
        return 15 + nextCondiment.getPrice();
    }
}
